Makefile

```diff
@@ -120,36 +120,19 @@ CONTROLLER_GEN = $(PROJECT_DIR)/bin/controller-gen
 controller-gen:
 	@GOBIN=$(PROJECT_DIR)/bin GO111MODULE=on $(GO_CMD) install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.10.0
 
-##### Devel image build and push targets #####
-.PHONY: .build-image .pull-build-image .push-build-image
-BUILDIMAGE ?= k8s-kata-manager-devel
-.build-image: Dockerfile.devel
-	if [ x"$(SKIP_IMAGE_BUILD)" = x"" ]; then \
-		$(DOCKER) build \
-			--progress=plain \
-			--build-arg GOLANG_VERSION="$(GOLANG_VERSION)" \
-			--tag $(BUILDIMAGE) \
-			-f $(^) \
-			.; \
-	fi
-
-.pull-build-image:
-	$(DOCKER) pull $(BUILDIMAGE)
-
-.push-build-image:
-	$(DOCKER) push $(BUILDIMAGE)
-
-$(DOCKER_TARGETS): docker-%: .build-image
-	@echo "Running 'make $(*)' in docker container $(BUILDIMAGE)"
+$(DOCKER_TARGETS): docker-%:
+	@echo "Running 'make $(*)' in container image $(BUILDIMAGE)"
 	$(DOCKER) run \
 		--rm \
-		-e GOCACHE=/tmp/.cache \
-		-v $(PWD):$(PWD) \
-		-w $(PWD) \
+		-e GOCACHE=/tmp/.cache/go \
+		-e GOMODCACHE=/tmp/.cache/gomod \
```

Comment:
I've noticed that we technically need to specify a golangci-lint cache here too.

Reply:
or we could run without docker, as in https://github.com/NVIDIA/k8s-device-plugin/blob/main/.github/workflows/golang.yaml#L42-L47

Reply:
Yes, but if I want to run `golangci-lint` after starting an interactive shell with `make .shell`.

Reply:
Can be done as a follow-up.